src/main/resources/com/google/api/codegen/java/test.snip

```diff
@@ -55,14 +55,23 @@
     {@setResponse(test.mockResponse)}
 
     {@initCode(test.initCode)}
+
+    {@setResponse(test.mockResponse)}
+
     {@methodCall(test)}
 
     List<GeneratedMessage> actualRequests = serviceHelper.getService().getRequests();
     Assert.assertEquals(1, actualRequests.size());
     {@test.requestTypeName} actualRequest = ({@test.requestTypeName})actualRequests.get(0);
 
-    @join assert : test.asserts
-      Assert.assertEquals(actualRequest.{@assert.actualValueGetter}(), {@assert.expectedValueIdentifier});
+    @join assert : test.requestAsserts
+      Assert.assertEquals({@assert.expectedValueIdentifier}, \
+        actualRequest.{@assert.actualValueIdentifier}());
+    @end
+
+    @join assert : test.responseAsserts
+      Assert.assertEquals(expectedResponse.{@assert.expectedValueIdentifier}(), \
+        actualResponse.{@assert.actualValueIdentifier}());
```

Comment:
Yes, sometimes the actual value needs to be an variable name but sometimes it could be a getter method name. I am trying to make these assert views as general as possible so they could fit for different usages. I changed the name to "actual" and "expected" so it does not indicate if it is a var or a method. Does it sound okay to you?